File: 169. Majority Element.py
from typing import List
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        nums.sort()
        n=len(nums)

        count=1
        i=1

        while i < n:
            if nums[i]==nums[i-1]:
                count+=1
            else:
                if count>n//2 :
                    break
                count=1
            i+=1

        
        return nums[i-1]

File: 169. Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int n=nums.size();
        int count=1;
        int i=1;
        for(;i<n;i++)
        {
            if(nums[i]==nums[i-1])
                count++;
            else
            {
                if(count>(n/2))
                    break;
                count=1;
            }
        }
        return nums[i-1];
    }
};